### DISCUSSION TITLE
Impove UI

### DISCUSSION THREAD

**Discussion by Catrya** [MostroP2P/mobile#26] [Ideas]
Ideas to improve the mobile client UI

**Comment by Catrya** [comment]
Hasta ahora tenemos aquí los diseños https://www.figma.com/design/Oh5qXO6pq12LbKPLw0NDPC/Mostro-UI?node-id=0-1&t=XQ7gecF5AZUWIlGN-0

El libro de órdenes está así
![Captura desde 2024-08-21 16-03-55](https://github.com/user-attachments/assets/bd404696-e451-4b00-981e-be21df6b9252)

Propongo algunos cambios: 
- Quitar los usuarios que publicaron las ofertas (la idea que se sepa quién la creó luego de tomar la orden, no antes)
- Añadir la bandera que le corresponda al fiat 
- Añadir las estrellas de la calificación
- Añadir los días usando mostro (si se decide poner, aún no está implementado)

Aquí un ejemplo de los cambios, no soy diseñadora así que seguro se puede mejorar mucho más:

![Captura desde 2024-08-21 16-04-02](https://github.com/user-attachments/assets/a4641380-87a7-49dd-aaf5-053770632e07)

También estaba pensando que se debería dejar que los métodos de pago fueran escritos por los usuarios, no seleccionados de una lista o algo así, solo en LATAM tenemos muchos métodos de pagos en cada país, así que mejor que cada uno lo escriba, porque tener que enumerarlos todos va a ser complicado.

**Comment by Catrya** [comment]
Luego de que el vendedor pague la hold invoice, darle un poco más de información de lo que pasará
ahora tenemos esto:
![Captura desde 2024-08-21 17-15-43](https://github.com/user-attachments/assets/0e04e018-0d90-44fa-ac9a-a19b9c06a9ae)

Propongo poner algo así:
![Captura desde 2024-08-21 17-13-42](https://github.com/user-attachments/assets/d5318ac2-0207-46b0-a077-419ed8843e87)

**Comment by Catrya** [comment]
Esta pantalla no va, pero una así se puede usar para el chat entre los 2 usuarios.
De la forma en que está es como si fuera mostro quien tomara los datos bancarios del vendedor para dárselos al comprador, y eso no sucede, sino que lo hablan ellos directamente, es cosa del cliente gestionar esa comunicación, los mostros no tienen acceso a la info personal que se intercambian los users

![Captura desde 2024-08-21 18-46-10](https://github.com/user-attachments/assets/4cc4bdee-2439-40a5-bcdf-93846fab681f)

**Comment by AndreaDiazCorreia** [comment]
Se entienden bien los cambios, una consulta, para la pantalla de loggeo tienen algun diseño? quieren que sea como la web app que aun sin iniciar se ve el market?

**Reply by Catrya** [reply]
Sí debe mostrar el market aunque no se esté logueado

Creo que no hay hecho un diseño para la pantalla de logueo. El del cliente web está bueno, aunque no sé si debería ser igual, no sé qué  creas, también @grunch y @bilthon a ver qué dicen, o quien se quiera sumar

**Reply by bilthon** [reply]
Si, las órdenes son públicas y no hay necesitad de poner ninguna barrera para que el usuario las vea.

**Reply by grunch** [reply]
El tema en mobile es el tamaño de la pantalla, tenemos que definir que debe pasar una vez el usuario ejecuta la app por primera vez 

Me refiero a si vamos a mostrar una pantalla que explique que es mostro y como funciona en 3 imágenes que el usuario va deslizando, algo breve un conciso, debemos definir que vamos a mostrar, esto se debe mostrar solo la primera vez (Andrea no hagas esto por ahora, estos son adornos que haremos luego)

luego nos debería llevar a una pantalla de login donde el usuario puede colocar su nsec o si prefiere no hacerlo puede pasar a la pantalla de listar órdenes pero obviamente sin posibilidad de crear órdenes ni tomarlas, solo para visualizar, la pantalla de login debería tener un botón para generar la nsec si el usuario no tiene

otro tema es como se le va a comunicar a los usuarios la información, soy partidario de usar un lenguaje más banal y no asustarlo hablando private keys, hay una discusión que tuve con maxi sobre la pantalla de registro, luego la busco y la pego acá (estoy desde la app de github)

**Reply by Catrya** [reply]
> Me refiero a si vamos a mostrar una pantalla que explique que es mostro y como funciona en 3 imágenes que el usuario va deslizando, algo breve un conciso, debemos definir que vamos a mostrar, esto se debe mostrar solo la primera vez (Andrea no hagas esto por ahora, estos son adornos que haremos luego)

esa idea me gusta, y que se quede en algún botón luego que se llame tutorial, o algo así, para si el usuario quiere volverlo a ver o enseñárselo a alguien más

**Reply by Catrya** [reply]
> otro tema es como se le va a comunicar a los usuarios la información, soy partidario de usar un lenguaje más banal y no asustarlo hablando claves privadas

de acuerdo, lo ideal sería que quien no quiera ni se entere que está usando Nostr, así que mejor no generar fricción para quienes nunca han usado nostr. tal vez luego se pueda poner enlace a la documentación para quienes quieran aprender más del funcionamiento

**Comment by Catrya** [comment]
@AndreaDiazCorreia, esto es lo que se me ocurre a la idea de @grunch de que le salga a los usuarios que puedan elegir si quieren usar una misma clave todo el tiempo (aparte de las efímeras) o si quieren que el cliente les genere una para cada intercambio. Es solo un prototipo, revisen porfa y vemos cómo se puede mejorar)

![Captura desde 2024-08-26 22-48-07](https://github.com/user-attachments/assets/2a466b17-5618-4a0e-ada7-5cba4c7e19f0)

La idea sería que saliera esa pantalla cuando un user abra la app por primera vez y vaya a tomar o crear una orden.  ¿Quizás en cuanto la abra por 1era vez sin esperar a que tome o cree una orden? Qué creen?
Tambíen en algún botón que sea settings para que se pueda cambiar

**Reply by AndreaDiazCorreia** [reply]
Esta genial, igual por ahora voy avanzando con las funcionalidades principales asi esos temas de UI los pueden seguir evaluando y se ajustan mas adelante
